llvm/lib/Transforms/InstCombine/InstCombineShifts.cpp

```diff
@@ -1206,6 +1206,12 @@ Instruction *InstCombinerImpl::visitShl(BinaryOperator &I) {
     return BinaryOperator::CreateAnd(Mask, X);
   }
 
+  // Transform  (-1 >> y) << y  to -1 << y
+  if (match(Op0, m_LShr(m_AllOnes(), m_Specific(Op1)))) {
```

Comment:
Any particular reason why this one uses m_LShr while the above uses m_Shr? It's valid for ashr as well: https://alive2.llvm.org/ce/z/fN5_7L

Reply:
`(ashr -1, x)` will always simplify to -1 on its own. Don't think adding support here for ashr would ever actually come into play